definitions.asm

```diff
@@ -7,103 +7,140 @@
+    .endvirtual
+
+    .endlogical
+cold_zp_table_end:
+
+
+; RAM System Variables
+
+; This table defines the initial values for forth user variables above zero page,
+; which are relocated to 'up' (defaulting to cp0) by COLD.
+
+cold_user_table:
+    .logical 0          ; this makes labels here offsets that we can add to 'up'
 
-; RAM System Variables: this must match cold_user_table in native_words.asm
 ; Block variables
-blk_offset = 0        ; BLK : UP + 0
-scr_offset = 2        ; SCR : UP + 2
+blk_offset:             .word 0         ; BLK
+scr_offset:             .word 0         ; SCR
 
 ; Wordlists
-current_offset = 4    ; CURRENT (byte) : UP + 4 (Compilation wordlist)
-num_wordlists_offset = 5
-                           ; #WORDLISTS (byte) : UP + 5
-wordlists_offset = 6  ; WORDLISTS (cells) : UP + 6 to UP + 29
-                           ;          (FORTH, EDITOR, ASSEMBLER, ROOT, +8 more)
-num_order_offset = 30 ; #ORDER (byte) : UP + 30
-                           ;          (Number of wordlists in search order)
-search_order_offset = 31
-                           ; SEARCH-ORDER (bytes) : UP + 31 to UP + 39
-                           ; Allowing for 9 to keep offsets even.
-max_wordlists = 12    ; Maximum number of wordlists supported
-                           ; 4 Tali built-ins + 8 user wordlists
+
+max_wordlists = 12    ; Maximum number of wordlists supported (4 built-in, 8 user wordlists)
+
+current_offset:         .byte 0         ; CURRENT = FORTH-WORDLIST (compilation wordlist)
+num_wordlists_offset:   .byte 4         ; #WORDLISTS (FORTH EDITOR ASSEMBLER ROOT)
+
+wordlists_offset:
+    .word dictionary_start              ; FORTH-WORDLIST
+    .word editor_dictionary_start       ; EDITOR-WORDLIST
+    .word assembler_dictionary_start    ; ASSEMBLER-WORDLIST
+    .word root_dictionary_start         ; ROOT-WORDLIST
+    .word 0,0,0,0,0,0,0,0               ; Space for 8 User wordlists
+
+num_order_offset:       .byte 1         ; #ORDER (Number of wordlists in search order)
+search_order_offset:
+    .byte 0,0,0,0,0,0,0,0,0             ; SEARCH-ORDER (9 bytes to keep offsets even)
 
 ; Buffer variables
-blkbuffer_offset    = 40   ; Address of buffer
-buffblocknum_offset = 42   ; Block number current in buffer
-buffstatus_offset   = 44   ; Status of buffer (bit 0 = used, bit 1 = dirty)
+
+blkbuffer_offset:       .word cp0+256   ; Address of buffer (right after USER vars)
+buffblocknum_offset:    .word 0         ; Block number current in buffer
+buffstatus_offset:      .word 0         ; Buffer status (bit 0 = used, bit 1 = dirty) (not in use)
 
 ; Block I/O vectors
-blockread_offset    = 46   ; Vector to block reading routine
-blockwrite_offset   = 48   ; Vector to block writing routine
+
+blockread_offset:       .word xt_block_word_error   ; Vector to block reading routine
+blockwrite_offset:      .word xt_block_word_error   ; Vector to block writing routine
+
+                        .word 0          ; 'COLD
```

Comment:
Do you know what this is for?  I couldn't find a reference to it in the code.

Reply:
I do not know where this came from.  BLAME shows you added it 3 weeks ago in [prune some code, add comments](https://github.com/SamCoVT/TaliForth2/commit/9db6ed4455663ff9e93b1c830814d2b774e21162)